src/Csv.php

```diff
@@ -777,8 +777,15 @@ public function unparse($data = array(), $fields = array(), $append = false, $is
         $entry = array();
 
         // create heading
+        $fieldOrder = $this->_validate_fields_for_unparse($fields);
+        if (!$fieldOrder && !empty($data)){
+            $column_count = count($data[0]) - 1;
```

Comment:
Judging by the variable name, the `- 1` should not be on this line, but within the `range()` call.